using_restricted_array.rb

```diff
@@ -12,10 +12,12 @@ def length(array)
   i = 0
   another_element = true
   while another_element
-    if array[i]
-      i += 1
-    else
-      another_element = false
+    if array[i] != SPECIAL_VALUE

```

Comment:
SPECIAL_VALUEs count towards length. They are just unused spaces. The size of an array doesn't change once created.